.buildkite/scripts/assert_docker_context_size.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+################################################################################
+# Ensure the Docker context remains a manageable, reasonable size.
+# (This helps catch missed renames that should be caught in the .dockerignore.)
+################################################################################
+# This can change in the future, but this seems like a reasonable starter limit
+readonly ARBITRARY_SIZE_LIMIT_MB=$((300))
+
+get_size_of_context_kb() {
+    # Based on https://stackoverflow.com/a/61821807
+    local -r context_dir="${1}"
+    temp_dir="$(mktemp --directory)"
+    (
+        cd "${context_dir}"
+        printf 'FROM scratch\nCOPY . /' | DOCKER_BUILDKIT=1 docker build \
+            --file - \
+            --output "${temp_dir}" \
+            .
+        echo "Dumped ${context_dir} context to ${temp_dir}" >&2
+        cd "${temp_dir}"
+        du --max-depth=0 | awk '{print $1;}'
+    )
+}
+
+# anything that'll be declared as a context in docker-bake.hcl
+all_docker_contexts() {
+    docker buildx bake all --print |
+        jq --raw-output '.target[].context' |
+        sort --unique
+}
+
+mapfile -t DIRS_TO_CHECK < <(all_docker_contexts)
+readonly DIRS_TO_CHECK
+
+failed_dirs=()
+
+for dir in "${DIRS_TO_CHECK[@]}"; do
+    echo "--- Checking Docker context size of ${dir}"
+    size_kb="$(get_size_of_context_kb "${dir}")"
+    echo "Docker context size of ${dir} is ${size_kb}KB"
```

Comment:
I was referring also to the intelligibility of the output generated here, where, say, "9000 MB" is a bit easier to mentally work with than "9000000 KB", particularly for the sizes we're going to be dealing with. Having such a long uninterrupted string of numbers is a bit difficult to read ("Is that 5 zeroes or 6?", etc.)

Looks like the easiest way to do that would be to add `--block-size=M` to the `du` command, but that appears to append the "M" to the output (and I can't see a way to _not_ do that with `du` options), requiring you to chop it off before you do your calculation.

Reply:
Right; string splitting and such I reeeeaaaally wanted to avoid doing in bash tbh